### DISCUSSION TITLE
Release v.21.0.0-R32

### DISCUSSION THREAD

**Discussion by jpsacha** [scalafx/scalafx#400] [General]
This release supports [JavaFX 21]. It was tested with Java 17 and Java 20. It can be used with Scala 2.12, 2.13, and Scala 3+.

Note that [JavaFX 21 requires Java **17 or newer**][JavaFX 21]. If you need to run on Java earlier than 17, please use previous releases of ScalaFX.

### New Features

* Support JavaFX 21 [[#395]]
    - Support new implementation of EventTarget [[#396]]
    - Add methods added to `javafx.application.Platform`: `canStartNestedEventLoop`
    - Add constants added to `scalafx.scene.AccessibleRole`: `CHECK_BOX_TREE_ITEM`
    - Add wrapper for "javafx.util.Subscription", new in JavaFX 21 [[#397]]
    - GridPane has a new constructor: GridPane(double hgap, double vgap) [[#399]]
* Move Scala 3 binary support to Scala 3.3 LTS [[#394]]

For downloads visit [ScalaFX on Sonatype][2] or [ScalaFX Releases][3] on GitHub. If you have ScalaFX related questions please use [ScalaFX Discussions][6], or [ScalaFX Users Group][5]. Please report any problems using [ScalaFX Issue Tracker][4].

[#394]: https://github.com/scalafx/scalafx/issues/394

[#395]: https://github.com/scalafx/scalafx/issues/395

[#396]: https://github.com/scalafx/scalafx/issues/396

[#397]: https://github.com/scalafx/scalafx/issues/397

[#399]: https://github.com/scalafx/scalafx/issues/399

[JavaFX 21]: https://openjfx.io/highlights/21/

[1]: http://scalafx.org

[2]: http://search.maven.org/#search&#124;ga&#124;1&#124;scalafx

[3]: https://github.com/scalafx/scalafx/releases

[4]: https://github.com/scalafx/scalafx/issues

[5]: https://groups.google.com/forum/#!forum/scalafx-users

[6]: https://github.com/scalafx/scalafx/discussions

[7]: https://stackoverflow.com/questions/tagged/scalafx

<hr /><em>This discussion was created from the release <a href='https://github.com/scalafx/scalafx/releases/tag/v.21.0.0-R32'>Release v.21.0.0-R32</a>.</em>

**Comment by objektwerks** [comment]
Well done, Jarek! Thanks! :)
